go/vt/worker/split_clone.go

```diff
@@ -622,8 +622,22 @@ func (scw *SplitCloneWorker) initShardsForVerticalSplit(ctx context.Context) err
 	}
 	sourceKeyspace := servedFrom
 
+	shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
+	shardMap, err := scw.wr.TopoServer().FindAllShardsInKeyspace(shortCtx, sourceKeyspace)
+	cancel()
+	if err != nil {
+		return fmt.Errorf("cannot find source shard for source keyspace %s: %s", sourceKeyspace, err)
+	}
+	if len(shardMap) != 1 {
+		return fmt.Errorf("found the wrong number of source shards, there should be only one, %v", shardMap)
+	}
+	var sourceShard string
+	for s := range shardMap {
+		sourceShard = s
+	}
```

Comment:
I think this ok for now. If we run into one more instance of code duplication, we'll have to make a function.